grails-app/controllers/org/broadinstitute/orsp/ReadOnlyInterceptor.groovy

```diff
@@ -0,0 +1,49 @@
+package org.broadinstitute.orsp
+
+import grails.converters.JSON
+
+/**
+ * Interceptor specific to API access related to the Read Only admin role
+ *
+ * Requires not to be ro_admin role to proceed
+ */
+class ReadOnlyInterceptor implements UserInfo {
```

Comment:
Can this work with [`@CompileStatic`](https://objectcomputing.com/resources/publications/sett/september-2015-grails-3-interceptors) ?

Reply:
No, I just tried and didn't work.

Reply:
👍

Reply:
@tadeork - I made the `@CompileStatic` change to Vero's branch that I'm testing and it seems to work fine.